### ISSUE TITLE
A11y: Wishlist icon for guests creates a change of context

### ISSUE THREAD

**Issue by ravenokko** [shopware/shopware#8188]
### Shopware Version

6.7

### Affected area / extension

Platform(Default)

### Actual behaviour

If you are a guest, and the Wishlist feature is active, there are cookie preferences you can accept which permit you, as a guest, to use the wishlist functionality. 

If you do not accept these preferences, clicking the wishlist icon on any product listing box/product detail page will redirect you to the sign up/login with no other warning or indication this will happen. Redirecting to a new page counts as a [change of context](https://www.w3.org/WAI/WCAG22/Understanding/on-input.html#dfn-changes-of-context) which is only appropriate when it is clear the user actions will result in such (for example: submitting a form, clicking a link). As the Wishlist icon is effectively a checkbox, this is a failure of [SC 3.2.2](https://www.w3.org/WAI/WCAG22/Understanding/on-input.html). 

### Expected behaviour

To Research
- do we really need the Wishlist enabled cookie? For legal or technical reasons?
**Update**: per LS-1158 we need to keep the Wishlist enabled option/cookie.

We need to adapt the design in this scenario asking the user to enable cookies so that guests have context why the wishlist doesn't work and what they need to do.

### How to reproduce

Activate the Wishlist feature. Click the heart icon either in a product detail page or product listing box as a guest in the storefront.
